src/main/java/com/celements/marshalling/Marshaller.java

```diff
@@ -0,0 +1,22 @@
+package com.celements.marshalling;
+
+import javax.annotation.concurrent.Immutable;
+import javax.validation.constraints.NotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+
+@Immutable
+public interface Marshaller<T> {
+
+  public @NotNull Class<T> getToken();
+
+  public @NotNull Object serialize(@NotNull T val);
```

Comment:
Replace Object with String. If a more general solution is needed (use case ?) then use generic S instead.